test/functional/morpheus_test.rb

```diff
@@ -0,0 +1,48 @@
+require 'test_helper'
+
+class MorpheusTest < ActionController::TestCase
+  tests ModelsController
+
+  include AuthenticatedTestHelper
+
+  test 'simulate model on morpheus button visibility' do
+    def morpheus_download_path(model)
+      blob = model.content_blobs.first
+      "morpheus://#{request.host_with_port}/models/#{model.id}/content_blobs/#{blob.id}/download"
+    end
+
+    public_model = FactoryBot.create(:morpheus_model, policy: FactoryBot.create(:public_policy))
+    teusink_model = FactoryBot.create(:teusink_jws_model, policy: FactoryBot.create(:public_policy))
+    private_model = FactoryBot.create(:morpheus_model, policy: FactoryBot.create(:private_policy))
+
+    # for a public Morpheus model, if Morpheus is not enabled, the button is hidden
+    with_config_value(:morpheus_enabled, false) do
+      get :show, params: { id: public_model }
```

Comment:
would be good to have each of these `get :show` requests and test in separate tests, to avoid problems when `rack` is updated to v3+